src/services/ganache/GanacheService.ts

```diff
@@ -83,8 +87,23 @@ export namespace GanacheService {
     return Promise.all(shouldBeFree).then(() => undefined);
   }
 
-  async function spawnGanacheServer(port: number | string): Promise<IGanacheProcess> {
-    const process = spawnProcess(undefined, "npx", [RequiredApps.ganache, `-p ${port}`]);
+  async function spawnGanacheServer(port: number | string, options?: TLocalProjectOptions): Promise<IGanacheProcess> {
+    const args: string[] = [RequiredApps.ganache, `--port ${port}`];
+
+    if (options?.isForked) {
+      if (options.url !== undefined) args.push(`--fork.url ${options.url}`);
+
+      if (
+        options.forkedNetwork !== undefined &&
+        options.forkedNetwork != Constants.treeItemData.service.local.type.forked.networks.other
+      )
+        args.push(`--fork.network ${options.forkedNetwork.toLowerCase()}`);
+
+      if (options.blockNumber !== undefined && options.blockNumber > 0)
+        args.push(`--fork.blockNumber ${options.blockNumber}`);
+    }
+
+    const process = spawnProcess(undefined, "npx", args);
     const output = window.createOutputChannel(`${Constants.outputChannel.ganacheCommands}:${port}`);
     const ganacheProcess = {port, process, output} as IGanacheProcess;
```

Comment:
at this point you can open the channel window with a command. 

https://code.visualstudio.com/api/references/vscode-api#OutputChannel

`/.show` command. You need a reference to your output channel, which you have just above.

Reply:
So cool! Thanks!